packages/app-store/routing-forms/pages/form-edit/[...appPages].tsx

```diff
@@ -91,43 +106,40 @@ function Field({
   const { t } = useLocale();
   const [animationRef] = useAutoAnimate<HTMLUListElement>();
 
-  const [options, setOptions] = useState<SelectOption[]>([
-    { placeholder: "< 10", value: "", id: uuidv4() },
-    { placeholder: "10-100", value: "", id: uuidv4() },
-    { placeholder: "100-500", value: "", id: uuidv4() },
-    { placeholder: "> 500", value: "", id: uuidv4() },
-  ]);
+  const options =
+    useWatch({
+      control: hookForm.control,
+      name: `${hookFieldNamespace}.options`,
+      defaultValue: [
+        { value: "", id: uuidv4() },
+        { value: "", id: uuidv4() },
+        { value: "", id: uuidv4() },
+        { value: "", id: uuidv4() },
+      ],
+    }) || [];
+
+  const setOptions = (updatedOptions: SelectOption[]) => {
+    hookForm.setValue(`${hookFieldNamespace}.options`, updatedOptions);
+  };
 
   const handleRemoveOptions = (index: number) => {
-    const updatedOptions = options.filter((_, i) => i !== index);
+    const updatedOptions = options.filter((option, i) => i !== index);
+    if (updatedOptions.length === 0) {
+      return;
+    }
     setOptions(updatedOptions);
-    updateSelectText(updatedOptions);
   };
 
   const handleAddOptions = () => {
-    setOptions((prevState) => [
-      ...prevState,
+    setOptions([
+      ...options,
       {
-        placeholder: "New Option",
         value: "",
         id: uuidv4(),
       },
     ]);
   };
 
-  useEffect(() => {
-    const originalValues = hookForm.getValues(`${hookFieldNamespace}.selectText`);
-    if (originalValues) {
-      const values: SelectOption[] = originalValues.split("\n").map((fieldValue) => ({
-        value: fieldValue,
-        placeholder: "",
-        id: uuidv4(),
-      }));
-      setOptions(values);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
```

Comment:
Removed this `useEffect` by doing this calculation at backend in `getSerializableForm`